lib/state/state.js

```diff
@@ -517,36 +511,72 @@ State.prototype.respond = function (response, isJson) {
 	this.response = isJson ? response : JSON.stringify(response);
 };
 
+function destroy(error, state, response, cb) {
+	// cleanup
+	state.destroy();
+
+	// return the response
+	if (cb) {
+		cb(null, response);
+	}
```

Comment:
I'm all for small functions, but this one does the opposite of making code more insightful:

- it does more than just destroy ... its second argument.
- it actually ignores the error argument.
- it only uses `response` and `cb`, just to combine the two (but the caller can't tell).
- it's only ever called from `response`, from where inlining the `destroy` function would have make things a lot clearer I think.